src/controllers/connection/newConnection.ts

```diff
@@ -200,15 +232,22 @@ export class NewConnectionController extends HTMLController {
   public async submitFromInvite(
     @Body()
     body: {
-      invite: BASE_64_URL
-      action: 'createConnection'
+      pin?: string
+      invite?: BASE_64_URL
+      action: 'createConnection' | 'pinSubmission'
     }
```

Comment:
We can type this better. For example 

```ts
body: {
  invite: BASE_64_URL
  action: 'createConnection'
} | {
  pin: string
  action: 'pinSubmission'
}
```

You can then use type guards to simplify other code